packages/express-backend/backend.js

```diff
@@ -1,19 +1,147 @@
 import express, { query } from "express";
 import cors from "cors";
+import multer from "multer";
 
 import userService from "./services/user-service.js";
 import chefService from "./services/chef-service.js";
 import { authenticateUser, registerUser, loginUser } from "./auth.js";
 import chefList from "./models/chefList.js";
 import Chef from "./models/chef.js";
 
+import {v2 as cloudinary} from 'cloudinary';
+          
+cloudinary.config({ 
+  cloud_name: 'dslmarna0', 
+  api_key: '743962474496839', 
+  api_secret: 'P8WYE5K596_PalkxT6DAGuyx6uE' 
+});
+
+async function handleUpload(file) {
+  const res = await cloudinary.uploader.upload(file, {
+    resource_type: "auto",
+  });
+  return res;
+}
+const storage = new multer.memoryStorage();
+const upload = multer({
+  storage,
+});
+
 
 const app = express();
 const port = 8000;
 
 app.use(cors());
 app.use(express.json());
 
+// gets the inputted form, stores base64 image in cloudinary and converts it 
+// to a url, stores that url in the database  
+app.post('/chefs', async (req, res) => {
+  try {
+    const {email, password, firstName, lastName, location, phoneNumber, cuisines, price, reviews, image, foodGallery} = req.body;
+    console.log("sent in json", req.body)
+    //console.log("image", image)
+    // Upload image to Cloudinary
+    let profilePicture;
+    if (image != null){
+    const uploadResponse = await cloudinary.uploader.upload(image, {
+      folder: 'chefs',
+      use_filename: true,
+      unique_filename: false,
+    });
+
+    profilePicture = uploadResponse.secure_url;
+  }
+  else{
+    profilePicture = 'noimage';
+  }
+    // Save chef data in the database (example using a mock database function)
+    const newChef = {
+      email,
+      password,
+      firstName,
+      lastName,
+      location,
+      phoneNumber,
+      cuisines,
+      price,
+      reviews,
+      profilePicture,
+      foodGallery
+    };
+    console.log("cheef", newChef)
+    // Mock function to save chef to the database
+    await chefService.addChef(newChef);
```

Comment:
is the mock function supposed to be permanent? was curious how it's different from app.post add chef

Reply:
the addChef function is adding it to the mongo database - this addChef function is in the chef-service.js file